client/checkout/platform-checkout/email-input-iframe.js

```diff
@@ -631,4 +603,28 @@ export const handlePlatformCheckoutEmailInput = async (
 			// do nothing, only respond to expected actions.
 		}
 	} );
+
+	if ( ! customerClickedBackButton ) {
+		// Check if user already has a WooPay login session.
+		if ( ! hasCheckedLoginSession ) {
+			openLoginSessionIframe( platformCheckoutEmailInput.value );
+		}
+	} else {
+		wcpayTracks.recordUserEvent(
+			wcpayTracks.events.PLATFORM_CHECKOUT_SKIPPED
+		);
+
+		searchParams.delete( 'skip_platform_checkout' );
+
+		let { pathname } = window.location;
+
+		if ( '' !== searchParams.toString() ) {
+			pathname += '?' + searchParams.toString();
+		}
+
+		history.replaceState( null, null, pathname );
+
+		// Safari needs to close iframe with this.
+		closeIframe( false );
```

Comment:
The `closeIframe` function here prevents the SMS OTP to be shown after going back from Safari as you mentioned on #5113, adding `closeLoginSessionIframe();` at the end of this code should resolve this problem.

Reply:
That does not seem to fix it. I tried to add a listener to popstate and beforeunload but they are not triggered